tests/surrogate_test.py

```diff
@@ -0,0 +1,201 @@
+        ChebyshevFirstKind.make_nested_set(2))
+    domain = [(-5, 5), (-8, 0)]
+    surrogate = Surrogate(domain, grid_generator)
+    grid_object = surrogate.grid
+    assert numpy.allclose(surrogate.domain,
+                          numpy.array(domain, ndmin=2))
+    assert surrogate.coefficients is None
+    assert isinstance(grid_object, IndexGrid)
+    assert surrogate.dimension == 2
+
+
+def test_error_grid_generator_type():
+    """Test grid generator's type."""
+    domain = [(-5, 5), (-8, 0)]
+    with pytest.raises(TypeError):
+        Surrogate(domain, 'not a grid generator')
+
+
+def test_map_function_1d():
+    """Test if points are transfromed properly."""
+    grid = SmolyakGridGenerator(ChebyshevFirstKind.make_nested_set(1))
+    domain = (-8, 12)
+    new_domain = (0, 1)
+    f = Surrogate(domain, grid)
+    assert f._mapdomain(2, domain, new_domain) == pytest.approx(0.5)
+    assert numpy.allclose(f._mapdomain([-3, 7],
+                                       domain, new_domain), [0.25, 0.75])
+
+
+def test_map_function_2d():
+    """Test if points are transfromed properly."""
+    grid = SmolyakGridGenerator(ChebyshevFirstKind.make_nested_set(1))
+    domain = ((-10, 10), (0, 2))
+    new_domain = ((0, 1), (-1, 1))
+    f = Surrogate(domain, grid)
+    assert numpy.allclose(f._mapdomain((-10, 0), domain, new_domain), [0, -1])
+    assert numpy.allclose(f._mapdomain([(0, 1), (10, 2)],
+                                       domain, new_domain), [[0.5, 0], [1, 1]])
+
+
+# evaluate function_0 at tests points for parametrization
+f_0_point_1 = function_0((0.649, 0, -0.9))
+f_0_point_2 = function_0((-0.885, 1, 0.275))
+
+
+@pytest.mark.parametrize('linear_solver, exact_values',
+                         [('lu', [f_0_point_1, f_0_point_2]),
+                          ('lstsq', [f_0_point_1, f_0_point_2]),
+                          ('inv', [f_0_point_1, f_0_point_2])])
+def test_surrogate_0(linear_solver, exact_values):
+    """Test if surrogate generates exact results for test fucntion 0."""
+    grid_gen = SmolyakGridGenerator(ChebyshevFirstKind.make_nested_set(2))
+    surrogate = Surrogate([(-1, 1), (-1, 1), (-1, 1)], grid_gen)
+    surrogate.train(function_0, linear_solver)
+    # random points in the domain
+    points = [(0.649, 0, -0.9), (-0.885, 1, 0.275)]
+    surrogate_values = [surrogate(x) for x in points]
+    assert numpy.allclose(surrogate_values, exact_values)
+
+
+def test_surrogate_1():
+    """Test if surrogate generates exact results for test for Chebyshevs."""
+    grid_gen = SmolyakGridGenerator(ChebyshevFirstKind.make_nested_set(2))
+    surrogate = Surrogate([(-1, 1), (-1, 1)], grid_gen)
+    surrogate.train(function_1, 'lstsq')
+    # random points in the domain
+    points = [(0.649, -0.9), (-0.885, 1)]
+    surrogate_values = [surrogate(x) for x in points]
+    exact_values = [function_1(x) for x in points]
+    assert numpy.allclose(surrogate_values, exact_values)
+
+
+def test_surrogate_0_shifted():
```

Comment:
I see how this could work for testing a shifted function, but I don't think it quite does enough. It would be good if you added a step where you evaluated both surrogates at effectively the same point (with shifting for `surrogate_2`) to make sure it actually evaluates the same.